adsmp/tasks.py

```diff
@@ -74,57 +78,76 @@ def task_route_record(bibcode, force=False, delayed=1):
-
+    logger.debug('Running after-update for: %s', bibcodes)
+    batch = []
+    batch_insert = []
+    batch_update = []
+    
     #check if we have complete record
-    r = app.get_record(bibcode)
-
-    if r is None:
-        raise Exception('The bibcode {0} doesn\'t exist!'.format(bibcode))
-
-    bib_data_updated = r.get('bib_data_updated', None)
-    orcid_claims_updated = r.get('orcid_claims_updated', None)
-    nonbib_data_updated = r.get('nonbib_data_updated', None)
-    fulltext_updated = r.get('fulltext_updated', None)
-    metrics_updated = r.get('metrics_updated', None)
-
-    year_zero = '1972'
-    processed = r.get('processed', adsputils.get_date(year_zero))
-    if processed is None:
-        # It was never sent to Solr
-        processed = adsputils.get_date(year_zero)
-
-    is_complete = all([bib_data_updated, orcid_claims_updated, nonbib_data_updated])
-
-    if is_complete:
-        if force is False and all([bib_data_updated and bib_data_updated < processed,
-               orcid_claims_updated and orcid_claims_updated < processed,
-               nonbib_data_updated and nonbib_data_updated < processed]):
-            logger.debug('Nothing to do, it was already indexed/processed')
-            return
-        else:
-            # build the record and send it to solr
-            logger.debug('Updating solr')
-            solr_doc = solr_updater.transform_json_record(r)
-            solr_updater.update_solr(solr_doc, app.conf.get('SOLR_URLS'))
-            app.update_processed_timestamp(bibcode)
-    else:
-        # if we have at least the bib data, index it
-        if force is True and bib_data_updated:
-            logger.warn('Forced indexing of: %s (metadata=%s, orcid=%s, nonbib=%s, fulltext=%s)' % \
-                        (bibcode, bib_data_updated, orcid_claims_updated, nonbib_data_updated, fulltext_updated))
-            # build the record and send it to solr
-            solr_doc = solr_updater.transform_json_record(r)
-            solr_updater.update_solr(solr_doc, app.conf.get('SOLR_URLS'))
-            app.update_processed_timestamp(bibcode)
+    for bibcode in bibcodes:
+        r = app.get_record(bibcode)
+    
+        if r is None:
+            raise Exception('The bibcode {0} doesn\'t exist!'.format(bibcode))
+    
+        bib_data_updated = r.get('bib_data_updated', None)
+        orcid_claims_updated = r.get('orcid_claims_updated', None)
+        nonbib_data_updated = r.get('nonbib_data_updated', None)
+        fulltext_updated = r.get('fulltext_updated', None)
+        metrics_updated = r.get('metrics_updated', None)
+    
+        year_zero = '1972'
+        processed = r.get('processed', adsputils.get_date(year_zero))
+        if processed is None:
+            # It was never sent to Solr
+            processed = adsputils.get_date(year_zero)
+    
+        is_complete = all([bib_data_updated, orcid_claims_updated, nonbib_data_updated])
+    
+        if is_complete:
```

Comment:
It seems we could re-order this "if" and have the if force in the external part, since if it is true it will be executed as far as there is bibdata and independently of any other condition.